From eb09957ffe8a5df640f34cee8986b979d34c90b2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20D=C3=B6singer?= <stefan@codeweavers.com>
Date: Thu, 4 Jul 2013 23:44:11 +0200
Subject: [PATCH 076/127] wined3d: Use double-buffered buffers for
 multithreaded CS
Reply-To: wine-devel <wine-devel@winehq.org>

---
 dlls/wined3d/buffer.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/dlls/wined3d/buffer.c b/dlls/wined3d/buffer.c
index dce9d8b..9d93d1e 100644
--- a/dlls/wined3d/buffer.c
+++ b/dlls/wined3d/buffer.c
@@ -1235,6 +1235,9 @@ static HRESULT buffer_init(struct wined3d_buffer *buffer, struct wined3d_device
     }
     buffer->maps_size = 1;
 
+    if (wined3d_settings.cs_multithreaded)
+        buffer->flags |= WINED3D_BUFFER_DOUBLEBUFFER;
+
     return WINED3D_OK;
 }
 
-- 
1.8.1.5

